sinonjs/referee-sinon
# Add failing test for issue with 2 samsam versions

#### Purpose (TL;DR) - mandatory

This demonstrates an issue that was discussed in sinonjs/sinon#2224. When using matchers created by one instance of samsam and using another instance of samsam to verify the matcher does not work.

#### Background (Problem in detail)  - optional

Matchers are detected using `match.isMatcher(obj)` here: https://github.com/sinonjs/samsam/blob/d2df9c28610d5d9f78719a4fea4aacce1d2c84ac/lib/deep-equal.js#L73

This check fails to detect a matcher because it's checking the prototype chain:

https://github.com/sinonjs/samsam/blob/d2df9c28610d5d9f78719a4fea4aacce1d2c84ac/lib/create-matcher/is-matcher.js#L14-L16

However, the prototype instances of two versions of the same module are different.

#### Solution  - optional

We need to make sure we're not shipping different versions of samsam in one package. This test will help to detect if this is the case in the future.

#### How to verify - mandatory
1. Check out this branch
2. `npm install`
3. `npm t`
4. Observe that the new test case fails unexpectedly.

#### Checklist for author
